### DISCUSSION TITLE
I am using p-treeselect but it is lagging when there is large data in  angular is there any alternative

### DISCUSSION THREAD

**Discussion by ghost** [primefaces/community#624] [PrimeNG] [answered]
<p-treeSelect
                        class=""
                        containerStyleClass="tree-select-wrapper w-100"
                        formControlName="attributes"
                        [options]="attribute"
                        display="chip"
                        [metaKeySelection]="false"
                        selectionMode="checkbox"
                        placeholder="Select Item"
                        [filter]="true"
                        [filterInputAutoFocus]="true"
                        [showClear]='true'
                      ></p-treeSelect> i am using this actually the problem is when the data is very large around 12000 of list then it is lagging and its response is very low

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for the feedback, this issue would be solved by virtualScroll support but treeSelect doesn't have it yet. We've added it in our internal backlog to support virtualScroll in treeSelect for the large data.

**Reply by ghost** [reply]
so is there any possible way

**Comment by sameer-saleem** [comment]
Hi,
I faced an issue when using TreeSelect with virtualScroll: the selected element would jump to the top whenever new items were loaded.

Adding this CSS resolved it for me:
```
.p-treeselect-items-wrapper {
  overflow-anchor: none;
}
```

This disables the browser’s scroll anchoring, which was causing the jump. After applying this, the selected element stays in place while new items load.

Hope it helps anyone facing the same problem!
